gamemodes/terrortown/entities/entities/ttt_beacon.lua

```diff
@@ -220,3 +224,23 @@ if CLIENT then
 		end
 	end)
 end
+
+---
+-- Hook that is called when a player is about to be found by a beacon.
+-- This hook can be used to cancel the detection.
+-- @param Player ply The player that the beacon has found
+-- @param Entity ent The beacon entity that found the player
+-- @return boolean Return false to cancel the player being detected
+-- @hook
+-- @realm client
```

Comment:
just FYI, as soon as markervision is merged, this will be also on the serverside